src/integTest/kotlin/io/snyk/SnykBulkFileListenerTest.kt

```diff
@@ -284,4 +305,139 @@ class SnykBulkFileListenerTest : BasePlatformTestCase() {
+        toolWindowPanel.getRootContainerIssuesTreeNode().add(ContainerImageTreeNode(issuesForImage, project))
+
+        getKubernetesImageCache(project)?.extractFromFile(addedPsiFile.virtualFile)
+    }
+
+    @Test
+    fun `test ContainerResults should drop cache and mark rescanNeeded when Container supported file CHANGED`() {
+        setUpContainerTest()
+        val psiFile = myFixture.configureByText("existing.yaml", TestYamls.podYaml())
+        createFakeContainerResultInCache(psiFile)
+
+        assertFalse(isContainerUpdateNeeded())
+
+        ApplicationManager.getApplication().runWriteAction {
+            PsiDocumentManager.getInstance(project).getDocument(psiFile)?.setText(TestYamls.podYaml() + "bla bla")
+        }
+        FileDocumentManager.getInstance().saveAllDocuments()
+
+        assertTrue(isContainerUpdateNeeded())
+        await().atMost(2, TimeUnit.SECONDS).until { containerCacheInvalidatedForFile(psiFile) }
+    }
+
+    @Test
+    fun `test ContainerResults should mark rescanNeeded when Container supported file CREATED`() {
+        setUpContainerTest()
+        createFakeContainerResultInCache()
+
+        assertFalse(isContainerUpdateNeeded())
+
+        myFixture.configureByText("new.yaml", TestYamls.podYaml())
+
+        assertTrue(isContainerUpdateNeeded())
+        assertFalse(
+            "None of Container file with issues should been marked as obsolete here",
+            containerCacheInvalidatedForFile(null)
+        )
+    }
+
+    @Test
+    fun `test ContainerResults should mark rescanNeeded when Container supported file COPIED`() {
+        setUpContainerTest()
+        val originalFile = myFixture.addFileToProject("existing.yaml", TestYamls.podYaml())
+        createFakeContainerResultInCache(originalFile)
+
+        assertFalse(isContainerUpdateNeeded())
+
+        lateinit var newFile : VirtualFile
+        ApplicationManager.getApplication().runWriteAction {
+            newFile = originalFile.virtualFile.copy(null, originalFile.virtualFile.parent, "copied.yaml")
+        }
+        myFixture.configureFromExistingVirtualFile(newFile)
+
+        assertTrue(isContainerUpdateNeeded())
+        assertFalse(
+            "None of Container file with issues should been marked as obsolete here",
+            containerCacheInvalidatedForFile(null)
+        )
+    }
+
+    @Test
+    fun `test ContainerResults should drop cache and mark rescanNeeded when Container supported file MOVED`() {
+        setUpContainerTest()
+        val originalFile = myFixture.addFileToProject("existing.yaml", TestYamls.podYaml())
+        createFakeContainerResultInCache(originalFile)
+
+        assertFalse(isContainerUpdateNeeded())
+
+        ApplicationManager.getApplication().runWriteAction {
+            val moveToDirVirtualFile = originalFile.virtualFile.parent.createChildDirectory(null, "subdir")
+            originalFile.virtualFile.move(null, moveToDirVirtualFile)
```

Comment:
Maybe an interesting case would also be a rename - the code should already support it, and it would be a good test to have.

Reply:
it's opened the rabbit hole of changes... but looks like we managed to reach the bottom of it... Thanks for catching that case!